source/extensions/tracers/zipkin/tracer.cc

```diff
@@ -33,6 +33,7 @@ SpanPtr Tracer::startSpan(const Tracing::Config& config, const std::string& span
   uint64_t random_number = random_generator_.random();
   span_ptr->setId(random_number);
   span_ptr->setTraceId(random_number);
+  span_ptr->setTraceIdHigh(random_generator_.random());
```

Comment:
so this always uses 128-bit? Probably fine by now, but might break some folks. Usually, we make this optional setting. ex https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/Tracing.java#L270

Reply:
PS this comment is about the implications of rollout. As-is, an old downstream service might break if they aren't prepared to consume 128-bit trace IDs. usually the behavior is to restart the trace, but not everything is consistent. That said, many libraries handle big IDs by now.

https://github.com/openzipkin/b3-propagation/issues/6

Reply:
cc @mattklein123 on this one..

Reply:
@adriancole Originally wasn't sure whether to make it configurable, as seems like 128 is fairly standard now - but there is potential for it causing disruption in a mesh if using different versions of envoy. So I will look to make it configurable, with default of 64 bit - but atleast the newer version will be able to handle a 128bit id even if it doesn't use it for new trace instances by default.